src/components/schedule-visualizer/robot-trajectories-overlay.tsx

```diff
@@ -26,6 +28,26 @@ export default function RobotTrajectoriesOverlay(
   // FIXME: hardcode for now, as the source of the footprint is expected to change.
   const footprint = 0.5;
 
+  useEffect(() => {
+    function getConflictRobotsName(): string[] {
```

Comment:
Maybe I'm not following you, I should have left a comment there. What that function does is the following

If this, for example, is our trajectory message
``` JSON
{
  "response":"trajectory",
  "conflicts": [310,311]
  "values":[
{
    "robot_name":"magni1",
    "fleet_name:"magni",
    "shape":"circle",
    "dimensions":0.3,
    "id":310,
    "segments":[
      {"t":336857,"v":[0.018886815933120995,0.4996431607843137,0.0],"x":[11.610485884950174,-8.054053944114406,-1.6085801124572754]},
      {"t":338700,"v":[0.01888681593558913,0.4996431608496359,0.0],"x":[11.645306833719362,-7.132879672805034,-1.6085801124572754]}]},

{
    "robot_name":"magni2",
    "fleet_name:"magni",
    "shape":"circle",
    "dimensions":0.3,
    "id":311,
    "segments":[
      {"t":336857,"v":[0.018886815933120995,0.4996431607843137,0.0],"x":[11.610485884950174,-8.054053944114406,-1.6085801124572754]},
      {"t":338700,"v":[0.01888681593558913,0.4996431608496359,0.0],"x":[11.645306833719362,-7.132879672805034,-1.6085801124572754]}]},

{id:312, ...},
],
}
```

Context:
We have these components in which the RobotTrajectoryOverlay pass their properties to its child RobotTrajectory (it receives just one trajectory).
`RobotTrajectoryOverlay -> RobotTrajectory`

`getConflictRobotsName` gets just the conflicting trajectories (310,311) and after that, gets the name of the robots inside each trajectory (values: [{},{}]). The mentioned function returns a string[] with the name of robots with conflicts.
After getting those values, the `useEffect` dispatches a message to the notification Bar to show the message. In this case, _RobotTrajectoryOverlay_ component is in charge of getting the conflict values and handle the notification. 

Once said that, Maybe what you are trying to said to me it's for me pass the `traj` prop that already contains the `robot_name` to the trajectory component and handle the message in the child component _RobotTrajectory_. There is a couple of issues here:
* I'll just have access to the name of the current trajectory on _RobotTrajectory_.
* I can trigger the Notification Bar from there with just one conflict trajectory.
* In case we want to RobotTrajectoryOverlay handle the Notification Bar just to show one message with all conflicts, we can send a dispatcher function from `RobotTrajectoryOverlay -> RobotTrajectory`. With this dispatcher, the _RobotTrajectory_ component would be able to create and modify a list of names of robots with conflicting paths. But I believe that will be unnecessary expensive because every RobotTrajectory should have to control one prop more and the logic inside RobotTrajectory will be replicated as many times as trajectories exist. 

``` typescript
 {trajs.map(traj => (
          <trajectoryContext.Component
            key={traj.id}
            trajectory={traj}
            footprint={footprint}
            conflicts={conflicts}
            colorManager={colorManager}
          />
        ))}

```


Hey sorry for writing so much, I just want to make sure that we are on the same page here.

Reply:
Thanks for the explanation! You are right, I missed that the conflict only contains the trajectory id so we still need to resolve that to the trajectory and get the robot name. The current way it's doing the notification looks good.